common/app/templates/inlineJS/blocking/googleAnalytics.scala.js

```diff
@@ -0,0 +1,35 @@
+@(page: model.Page)
+@import implicits.Strings.string2encodings
+
+var gaRandomParticipation = Math.floor(Math.random() * 100) < @GoogleAnalyticsAccount.samplePercent;
+var gaParticipation = gaRandomParticipation ? 'in' : 'out';
+
+try {
+    @*
+    Once a person is in the test group we want them to stay in.
+    We are more interested in "users" than in raw page views.
+    *@
+    if (window.localStorage) {
+        var gaStorageKey = 'gu.ga.participation';
+        gaParticipation = window.localStorage.getItem(gaStorageKey) || gaParticipation;
+        window.localStorage.setItem(gaStorageKey, gaParticipation);
+    }
+} catch (ex) {
+    @* if for some reason we cannot read/write to localStorage don't be in the test *@
+    gaParticipation = 'out';
```

Comment:
it's not strictly an error, it's more of a safe way to feature detect – does it need an error message?